pennylane_lightning/src/simulators/lightning_qubit/measurements/tests/Test_MeasurementsLQubitSparse.cpp

```diff
@@ -54,66 +54,42 @@ TEMPLATE_PRODUCT_TEST_CASE("Expected Values - Sparse Hamiltonian [Kokkos]",
 
-    if constexpr (USE_KOKKOS) {
-        SECTION("Testing Sparse Hamiltonian:") {
-            long num_qubits = 3;
-            long data_size = Pennylane::Util::exp2(num_qubits);
-
-            std::vector<long> row_map;
-            std::vector<long> entries;
-            std::vector<ComplexT> values;
-            write_CSR_vectors(row_map, entries, values, data_size);
-
-            PrecisionT exp_values = Measurer.expval(
-                row_map.data(), static_cast<long>(row_map.size()),
-                entries.data(), values.data(),
-                static_cast<long>(values.size()));
-            PrecisionT exp_values_ref = 0.5930885;
-            REQUIRE(exp_values == Approx(exp_values_ref).margin(1e-6));
-
-            PrecisionT var_values =
-                Measurer.var(row_map.data(), static_cast<long>(row_map.size()),
-                             entries.data(), values.data(),
-                             static_cast<long>(values.size()));
-            PrecisionT var_values_ref = 2.4624654;
-            REQUIRE(var_values == Approx(var_values_ref).margin(1e-6));
-        }
-
-        SECTION("Testing Sparse Hamiltonian (incompatible sizes):") {
-            long num_qubits = 4;
-            long data_size = Pennylane::Util::exp2(num_qubits);
-
-            std::vector<long> row_map;
-            std::vector<long> entries;
-            std::vector<ComplexT> values;
-            write_CSR_vectors(row_map, entries, values, data_size);
-
-            PL_CHECK_THROWS_MATCHES(
-                Measurer.expval(row_map.data(),
-                                static_cast<long>(row_map.size()),
-                                entries.data(), values.data(),
-                                static_cast<long>(values.size())),
-                LightningException,
-                "Statevector and Hamiltonian have incompatible sizes.");
-        }
-    } else {
-        SECTION("Testing Sparse Hamiltonian:") {
-            long num_qubits = 3;
-            long data_size = Pennylane::Util::exp2(num_qubits);
-
-            std::vector<long> row_map;
-            std::vector<long> entries;
-            std::vector<ComplexT> values;
-            write_CSR_vectors(row_map, entries, values, data_size);
-
-            PL_CHECK_THROWS_MATCHES(
-                Measurer.expval(row_map.data(),
-                                static_cast<long>(row_map.size()),
-                                entries.data(), values.data(),
-                                static_cast<long>(values.size())),
-                LightningException,
-                "Executing the product of a Sparse matrix and a vector needs "
-                "Kokkos and Kokkos Kernels installation.");
-        }
+    SECTION("Testing Sparse Hamiltonian:") {
+        long num_qubits = 3;
+        long data_size = Pennylane::Util::exp2(num_qubits);
+
+        std::vector<long> row_map;
+        std::vector<long> entries;
+        std::vector<ComplexT> values;
```

Comment:
I know this has been answered before, but do we need to used signed representations for these?

Reply:
The `CrsMatrix` class def starts with 
```
class CrsMatrix {
  static_assert(
      std::is_signed<OrdinalType>::value,
      "CrsMatrix requires that OrdinalType is a signed integer type.");
```
If I just comment the `static_assert`, everything compiles and tests pass. Wonder why this is there ...

Reply:
This may be a carry-over to L-GPUs sparse type, which requires signed? Or maybe a BLAS interface requires it? I'm fine to use it if it makes sense to do so.

Reply:
`CrsMatrix` is a class defined in Kokkos-kernels, so it is a limitation introduced by them. If we get rid of Kokkos-kernels (and all the building and testing related to it) we can get rid of this constraint. Up for that? @AmintorDusko too?

Reply:
Yes, Kokkos does not accept unsigned integers here. I'm OK with removing it if this is not needed anymore.

Reply:
Yep, if we don't need it, let's destroy it.

Reply:
:fire: